### ISSUE TITLE
Split up files with `// ignore-tidy-filelength`

### ISSUE THREAD

**Issue by varkor** [rust-lang/rust#60302]
These files are over 3,000 lines, which is not ideal for navigating or comprehending. If we can, it would be good to split these up. See also https://github.com/rust-lang/rust/issues/60015, which is a specific issue of this problem.

**Comment by petrochenkov** [comment]
https://github.com/rust-lang/rust/issues/61097 is going to help with some of the files.

**Comment by olamiko** [comment]
Which files @varkor? The files in #60015 are done

**Comment by varkor** [comment]
I count [17 different files](https://github.com/rust-lang/rust/search?q=%2F%2F+ignore-tidy-filelength&unscoped_q=%2F%2F+ignore-tidy-filelength) that still have this comment.

**Comment by alex-griffiths** [comment]
Hey, I'm not sure if anyone is working on this, but, this looks like it might be a good place to get started contributing.

I am quite new to the language, but I'd like to get involved. Is this something as simple as pulling out structs and other related blocks of code into their own files and then including them in the original?

**Comment by varkor** [comment]
@alex-griffiths: yes, it should be as simple as figuring out some sensible way to split up large files into related types and implementations!

**Comment by alex-griffiths** [comment]
Alright, mind if I take a crack at this?

**Comment by varkor** [comment]
@alex-griffiths: yes, go ahead! You can claim the issue with:
> @rustbot claim

**Comment by alex-griffiths** [comment]
@varkor I'm finding that a lot of the files that include the `// ignore-tidy-filelength` comment are files that include lots of the documentation.

I'm still kind of new to the language, but for example in src/libstd/fs.rs there are 6 structs where the struct definition itself is a single line, but with varying amounts of docs. Is this the kind of thing that is able to be extracted into it's own file which is then included as a module in the fs.rs file?

**Comment by varkor** [comment]
@alex-griffiths: that's an interesting observation. I think in these cases, it's not so important to split up the files, because the length doesn't pose an issue for comprehension. Anyone editing the file is going to be interested in one or two particular functions, which are well-isolated within the file, and wouldn't benefit from being in their own file. In some sense, files like these are "false positives", though I don't think we could automatically exclude them. The lint is really more for files with large functions, and interdependency that we want to make easier to navigate. Thanks for pointing this out!

**Comment by alex-griffiths** [comment]
I'll start working through all the files that have the comment in them, and I'll compile a list of files that appear to mostly documentation, and then for the rest, I'll start trying to improve the navigability of them.

**Comment by Daniihh** [comment]
Would like to help with this, too. Is there a possibility to share an issue?
Specifically, I have my eyes on [core::num](https://github.com/rust-lang/rust/blob/master/library/core/src/num/mod.rs).

**Comment by jyn514** [comment]
Seems like a good idea. You need
- a list of the files to be split up
- instructions for how to split them up (more detailed is better, but simple is fine as long as you update it along the way)
- someone to keep the tracking issue up to date as people claim files. Maybe in this case that would be @varkor?

**Comment by Daniihh** [comment]
Oh sweet, thanks! This'll be my first contributions to the rust standard library. 😄
I personally don't think we'll need instructions on splitting the files up as long as everyone's respectful. Although, it is @alex-griffiths decision.
If no one minds I'm gonna list out the files that need truncating.

**Comment by Daniihh** [comment]
- [src/test/ui/issues/issue-29466.rs](https://github.com/rust-lang/rust/blob/master/src/test/ui/issues/issue-29466.rs)
- [src/test/ui/issues/issue-72933-match-stack-overflow.rs](https://github.com/rust-lang/rust/blob/master/src/test/ui/issues/issue-72933-match-stack-overflow.rs)
- [alloc::vec](https://github.com/rust-lang/rust/blob/master/library/alloc/src/vec.rs)
- [core::tests::iter](https://github.com/rust-lang/rust/blob/master/library/core/tests/iter.rs)
- [rustc_resolve](https://github.com/rust-lang/rust/blob/master/compiler/rustc_resolve/src/lib.rs)
- [core::iter::traits::iterator](https://github.com/rust-lang/rust/blob/master/library/core/src/iter/traits/iterator.rs)
- [rustc_middle::ty](https://github.com/rust-lang/rust/blob/master/compiler/rustc_middle/src/ty/mod.rs)
- [compiletest::runtest](https://github.com/rust-lang/rust/blob/master/src/tools/compiletest/src/runtest.rs)
- [alloc::collections::vec_deque](https://github.com/rust-lang/rust/blob/master/library/alloc/src/collections/vec_deque.rs)
- [src/test/ui/issues/issue-74564-if-expr-stack-overflow.rs](https://github.com/rust-lang/rust/blob/master/src/test/ui/issues/issue-74564-if-expr-stack-overflow.rs)
- [core::num](https://github.com/rust-lang/rust/blob/master/library/core/src/num/mod.rs)
- [core::str](https://github.com/rust-lang/rust/blob/master/library/core/src/str/mod.rs)
- [rust_doc::html::render](https://github.com/rust-lang/rust/blob/master/src/librustdoc/html/render/mod.rs)
- [rustc_apfloat::tests::ieee](https://github.com/rust-lang/rust/blob/master/compiler/rustc_apfloat/tests/ieee.rs)
- [core::slice](https://github.com/rust-lang/rust/blob/master/library/core/src/slice/mod.rs)
- [rustc_typeck::check](https://github.com/rust-lang/rust/blob/master/compiler/rustc_typeck/src/check/mod.rs)

**Comment by alex-griffiths** [comment]
I believe changes to rustc_resolve need to be handled in a different manner.
https://github.com/rust-lang/rust/pull/75701#issuecomment-676266089

**Comment by lzutao** [comment]
I am doing core::slice splitting.

**Comment by lzutao** [comment]
Hi @alex-griffiths, Would you mind if I take care of `core::str` part ?

**Comment by jyn514** [comment]
@lzutao maybe do `core::num` first since alex is in the middle of splitting `core::str`

**Comment by lzutao** [comment]
Yeah, he also did `alloc/vec.rs` in #75701 . But he seems to having trouble in splitting `core/str.rs`.
I would like to help him a hand.

**Comment by lzutao** [comment]
I'm doing core::num.

**Comment by Daniihh** [comment]
I've finally got time, guess I'll be doing core::tests::iter.
Or is it core::tests::iter? I checked and apparently it's in another target other than lib, and I don't know how to represent that, lol.

Draft: #76391

**Comment by nixphix** [comment]
I ll work on [rustc_apfloat::tests::ieee](https://github.com/rust-lang/rust/blob/master/compiler/rustc_apfloat/tests/ieee.rs)

**Comment by lzutao** [comment]
Is there any reviewers want to review big PR: #76327 ? 
If not, I'm going to close it if it'd have merge conflicts again.

**Comment by Nicholas-Baron** [comment]
I will try my hand at [rustc_typeck::check](https://github.com/rust-lang/rust/blob/master/compiler/rustc_typeck/src/check/mod.rs).

**Comment by Nicholas-Baron** [comment]
@Daniihh The issue is no longer with [rustc_typeck::check](https://github.com/rust-lang/rust/blob/master/compiler/rustc_typeck/src/check/mod.rs), but with [rustc_typeck::check::fn_ctxt](https://github.com/rust-lang/rust/blob/master/compiler/rustc_typeck/src/check/fn_ctxt.rs). 

The `FnCtxt` struct that it houses has a lot of functionality wrapped into it (the `struct {}` alone taking 85 lines w/documentation). 

The more permanent solution to its length issue is to (in my opinion) move some fields out to "sub-structs".

**Comment by lzutao** [comment]
Could somebody review #76325 ? It stands still 19+ days without review.

**Comment by Nicholas-Baron** [comment]
@danii My pull request for `FnCtxt` was merged. Take my name off the list, please.

**Comment by DeveloperC286** [comment]
@varkor Do you or someone want to take a look at my PR https://github.com/rust-lang/rust/pull/78590 it has been 11 days and no comments so far. Thanks,

**Comment by Nicholas-Baron** [comment]
I am going to try and land `rustdoc::render`. Moving `Context` and `SharedContext` to separate files seems to be a starting point.

**Comment by Nicholas-Baron** [comment]
@danii PR #82088 was successfully merged and shortened `rustdoc::html::render` to ~2500 lines.

**Comment by Nicholas-Baron** [comment]
@danii PR #82964 opened to shorten `rustc_middle::ty`.

**Comment by Nicholas-Baron** [comment]
@danii PR #82964 merged.

**Comment by jmj0502** [comment]
Hey! I would like to start contributing to the project and this seems like a good starting point. Is the `compiletest::runtest` file still available?

**Comment by workingjubilee** [comment]
Yes, go ahead.

**Comment by Sam-Buckley** [comment]
I'd like to start contributing, and was directed here as a good starting point. What's still outstanding in this issue?
@rustbot claim

**Comment by workingjubilee** [comment]
@Sam-Buckley I updated the top of this issue with the current list: https://github.com/rust-lang/rust/issues/60302#issue-437574254

**Comment by jieyouxu** [comment]
Updated issue description to not include `compiletest::runtest`, which is tracked separately in https://github.com/rust-lang/rust/issues/89475.

**Comment by Sam-Buckley** [comment]
Sorry to waste everyone's time, family matters came up and I'll have to stop for a while

**Comment by ranger-ross** [comment]
Hey there, I'll take a crack at cleaning some of these large files up.

@rustbot claim

**Comment by NoobProgrammer31** [comment]
@rustbot claim


assign this to me i will work upon it

**Comment by NoobProgrammer31** [comment]
can i start working on mod.rs first in first file ( i am new to it )

**Comment by jieyouxu** [comment]
You don't need to finish all of them in one PR and please do not break all of them up in one PR. You can tackle each of the large files in a standalone PR.
